docs/HowToFAQ/Errors-Rewriter.md

```diff
@@ -0,0 +1,343 @@
+the same indentation. This warning occurs when the observed indentation is likely to cause
+a misunderstanding of the code, particularly of the precedence of operations. It is suggested
+to use parentheses or to redo the indentation to make the structure of the expression clearer.
+
+## **Warning: unusual indentation in _what_ (which ends at _location_); do you perhaps need parentheses?**
+
+<!-- %check-resolve-warn -->
+```dafny
+function f(b: bool): bool
+{
+  b ==> b && 
+  b          // this parses as b ==> (b && b), but reads as (b ==> b) && b
+}
+```
+
+Long expressions are best split up across multiple lines; readers often use indentation as a guide
+to the nesting of subexpressions. For example, parallel conjuncts or disjuncts may be written with 
+the same indentation. This warning occurs when the observed indentation is likely to cause
+a misunderstanding of the code, particularly of the precedence of operations. It is suggested
+to use parentheses or to redo the indentation to make the structure of the expression clearer.
+
+
+<!-- ./DafnyCore/Resolver/RunAllTestsMainMethod.cs -->
+
+## **Error: Cannot use /runAllTests on a program with a main method**
+
+<!-- TODO - this won't be triggered until Issue 3381 is fixed -->
+
+This error message when using `dafny test` and in the situation
+where a Main method is already present in the program.  
+`dafny test` synthesizes a Main method that calls all the test routines,
+which then conflicts with the existing Main method.
+(The `/runAllTests` option has been replaced by `dafny test`.)
+
+## **Error: Methods with the {:test} attribute can have at most one return value**
+
+<!-- %check-test -->
+```dafny
+method {:test} m(i: int) returns (j: int, k: int) 
+{
+  j, k := 0,0;
+}
+```
+
+This error only occurs when using `dafny test`. That command executes all methods attributed
+with `{:test}`, but such test methods are limited to methods with only one output parameter.
+This is purely an implementation limitation.
+
+
+<!-- ./DafnyCore/Resolver/ExpectContracts.cs-->
+
+## **Warning: The _kind_ clause at this location cannot be compiled to be tested at runtime because it references ghost state.**
+<!-- TODO - requires a companion target program file to run successfully -->
+<!-- %no-check %check-test %options --test-assumptions:Externs -->
+```dafny
+method {:extern} m(i: int, ghost j: int) 
+  requires j == 1
+```
+
+The `--test-assumptions` option inserts tests of various contracts and implicit assumptions 
+in a Dafny method, such as the requires and ensures clauses.
+However, because these inserted tests are compiled into the target
+program as runtime checks, they cannot refer to any ghost variables.
+(This mechanism implements Dafny's runtime-assertion checking.)
+ 
+## **Warning: Internal: no wrapper for _name_**
+
+This message indicates a bug within dafny. Please report the program that
```

Comment:
```suggestion
This message indicates a bug within Dafny. Please report the program that
```

Reply:
Dafny is the language; dafny is the tool. Reworded to be clearer.